#[130385341] Taskers should be able to submit service requests

alex-magana opened the pull request:
#### What does this PR do?

Create a table `service`
Create a model and controller for the table `service`
Create a view to capture service request details
Create a view to display requests that are yet to be assigned to an artisan
#### What are the relevant pivotal tracker stories?

130385341
#### Screenshots (if appropriate)

<img width="1680" alt="screen shot 2016-10-28 at 7 51 56 pm" src="https://cloud.githubusercontent.com/assets/20963836/19814750/530b0c18-9d48-11e6-92fe-d4ec280f0f6e.png">
<img width="1680" alt="screen shot 2016-10-28 at 7 53 33 pm" src="https://cloud.githubusercontent.com/assets/20963836/19814764/5ff44b74-9d48-11e6-9f82-5915b98eb5a0.png">
<img width="1680" alt="screen shot 2016-10-28 at 7 53 43 pm" src="https://cloud.githubusercontent.com/assets/20963836/19814772/6cac93e4-9d48-11e6-9a35-5ba025d20ad9.png">

alex-magana committed:
feat(services): add line to import services CSS file

alex-magana committed:
feat(services): add association between Skillset and Service

specify a has_many association following the one-to-many
relationship between model Skillset and model Service.

alex-magana committed:
feat(services): add association between User and Service

alex-magana committed:
feat(services): add links for 'Submit Request' and 'Pending Requests'

alex-magana committed:
feat(services): expose routes for ServicesController actions

alex-magana committed:
feat(services): add definition for table 'services'

alex-magana committed:
feat(services): add helper for testing request submission process

alex-magana committed:
feat(services): add asset files for model 'Service'

alex-magana committed:
feat(services): add controller for model 'Service'

alex-magana committed:
feat(services): add helpers, model and views files for model Service

alex-magana committed:
feat(services): define migration for table 'services'

alex-magana committed:
feat(services): remove column is_assigned from table 'services'

alex-magana committed:
feat(services): add column status to table 'services'

alex-magana committed:
feat(services): set default value for column status as 0

alex-magana committed:
feat(services): add tests for controller ServicesController

alex-magana committed:
feat(services): add factory for model Service

alex-magana committed:
feat(services): add feature tests for request submission

alex-magana committed:
feat(services): add model tests for model Service

alex-magana committed:
Merge branch 'master' into ft-request-submission-system-130385341

alex-magana committed:
feat(services): limit request submission to user tasker

alex-magana committed:
feat(services): correct schema file version

alex-magana committed:
feat(services): remove incorrect data in seed file

alex-magana committed:
feat(services): replace have_content with count

assert 'Pending Requests' view is rendered by checking the number
of links 'VIEW' displayed on the view.

alex-magana committed:
feat(services): add invocation of service allocation module

alex-magana committed:
feat(services): remove unused helper file

andela-fadebayo commented on app/assets/stylesheets/services.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Services controller here.
```
Remove this file

andela-fadebayo commented on app/models/service.rb:
```diff
@@ -0,0 +1,52 @@
+
+  def self.pending_requests(current_user)
+    where("tasker_id = ? AND status = ?", current_user.id, 0)
+  end
```
Use `scope` instead

andela-fadebayo commented on app/views/services/show.html.erb:
```diff
@@ -0,0 +1,46 @@
+
+              <div class="row">
+                <strong>Duration:</strong><br>
+                <%= @service.duration %>
```
Display duration as integers in days.

alex-magana committed:
feat(services): refactor the service feature

remove the unnecessary services file. remove unnecessary code comments.
display duration as an integer.

alex-magana committed:
feat(services): use do..end block for multiline block

alex-magana committed:
feat(services): replace class method with scope

alex-magana committed:
feat(services): add id attribute to link in pending requests

alex-magana committed:
feat(services): find link in pending requests by xpath
